EIPS/eip-1474.md

```diff
@@ -0,0 +1,2314 @@
+
+#### Error codes
+
+If an Ethereum RPC method encounters an error, the `error` member included on the response object **MUST** be an object containing a `code` member and descriptive `message` member. The following list contains all possible error codes and associated messages:
+
+|Code|Message|Meaning|Category|
+|-|-|-|-|
+|-32700|Parse error|Invalid JSON|standard|
+|-32600|Invalid request|JSON is not a valid request object|standard|
+|-32601|Method not found|Method does not exist|standard|
+|-32602|Invalid params|Invalid method parameters|standard|
+|-32603|Internal error|Internal JSON-RPC error|standard|
+|-32000|Invalid input|Missing or invalid parameters|non-standard|
+|-32001|Resource not found|Requested resource not found|non-standard|
+|-32002|Resource unavailable|Requested resource not available|non-standard|
+|-32003|Transaction rejected|Transaction creation failed|non-standard|
+|-32004|Method not supported|Method is not implemented|non-standard|
+
+Example error response:
+
+```sh
+{
+    "id": 1337
+    "jsonrpc": "2.0",
+    "error": {
+        "code": -32003,
+        "message": "Transaction rejected"
+    }
+}
+```
+
+#### Value encoding
+
+Specific types of values passed to and returned from Ethereum RPC methods require special encoding:
+
+##### `Quantity`
+
+- A `Quantity` value **MUST** be hex-encoded.
+- A `Quantity` value **MUST** be "0x"-prefixed.
+- A `Quantity` value **MUST** be expressed using the fewest possible hex digits per byte.
+- A `Quantity` value **MUST** express zero as "0x0".
+
+Examples `Quantity` values:
+
+|Value|Valid|Reason|
+|-|-|-|
+|0x|`invalid`|empty not a valid quantity|
+|0x0|`valid`|interpreted as a quantity of zero|
+|0x00|`invalid`|leading zeroes not allowed|
+|0x41|`valid`|interpreted as a quantity of 65|
+|0x400|`valid`|interpreted as a quantity of 1024|
+|0x0400|`invalid`|leading zeroes not allowed|
+|ff|`invalid`|values must be prefixed|
+
+
+##### `Data`
+
+- A `Data` value **MUST** be hex-encoded.
+- A `Data` value **MUST** be "0x"-prefixed.
+- A `Data` value **MUST** be expressed using two hex digits per byte.
+
+Examples `Data` values:
+
+|Value|Valid|Reason|
+|-|-|-|
+|0x|`valid`|interpreted as empty data|
+|0x0|`invalid`|each byte must be represented using two hex digits|
+|0x00|`valid`|interpreted as a single zero byte|
+|0x41|`true`|interpreted as a data value of 65|
+|0x004200|`true`|interpreted as a data value of 16896|
+|0xf0f0f|`false`|bytes require two hex digits|
+|004200|`false`|values must be prefixed|
+
+##### Proposing changes
+
+New Ethereum RPC methods and changes to existing methods **MUST** be proposed via the traditional EIP process. This allows for community consensus around new method implementations and proposed method modifications. RPC method proposals **MUST** reach "draft" status before being added to this proposal and the official Ethereum RPC specification defined herein.
```

Comment:
s/traditional//

Reply:
Also, once this standard is final, new methods shouldn't be added here - they should be added in new, separate EIPs, which undergo their own standardisation process.